executor/builder.go

```diff
@@ -1189,14 +1200,14 @@ func (b *executorBuilder) getStartTS() (uint64, error) {
 	}
 
 	startTS := b.ctx.GetSessionVars().SnapshotTS
-	if startTS == 0 && b.ctx.Txn(true).Valid() {
-		startTS = b.ctx.Txn(true).StartTS()
+	txn, err := b.ctx.Txn(true)
+	if err != nil {
+		return 0, errors.Trace(err)
 	}
-	b.startTS = startTS
-	if b.startTS == 0 {
```

Comment:
why remove this check?

Reply:
Because the error would be catch at line 1209
https://github.com/pingcap/tidb/pull/8712/files#diff-badfcd30d7596a08cd207b8e6ae778e6R1209